src/app/components/navbar/navbar.component.html

```diff
@@ -10,8 +10,12 @@
         <a class="nav-link" [routerLink]="['/playlists']" routerLinkActive="active">Playlists</a>
       </div>
     </div>
-    <div class="navbar-nav">
+    <div *ngIf="!auth.currentUser" class="navbar-nav">
       <a class="nav-link" [routerLink]="['/sign-up']">Sign Up</a>
     </div>
+    <div class="navbar-nav">
+      <a *ngIf="!auth.currentUser" class="nav-link" [routerLink]="['/log-in']">Log In</a>
+      <a *ngIf="auth.currentUser" class="nav-link" (click)="logOut()" [routerLink]="['/tracks']">Log Out</a>
```

Comment:
Please remove [routerLink]="['/tracks']";
As far as I can see you are doing redirect in logOut function